Switch to W3C naming for dictionary keys

pmeenan opened the pull request:
Use full name for the dictionary keys to match W3C guidance on naming.

We may need to shorten them back as part of the IETF process but 3 characters in the default config of `p` to `path` or 9 when including `e` to `expires` isn't the end of the world.

Fix #27

pmeenan commented:
FYI @horo-t

horo-t commented:
lgtm

Created a Chromium side change: https://chromium-review.googlesource.com/c/chromium/src/+/4456368

yoavweiss reviewed (approved):
LGTM

yoavweiss commented:
Given the use of wildcards, `match` may be better indeed.

pmeenan commented:
I switched to `match` instead of `path` and changed the wording through the doc to make it clear that it is a URL-matching pattern (instead of all of the path language) and linked it directly to the existing URL spec and processing (the wildcard doesn't change the parsing or relative URL expansion bits).

I also fixed a few examples that were still using prefix matching instead of wildcards and updated the language in the examples to match the fully-expanded URL patterns.

horo-t commented:
Still lgtm. Thank you.
Updated the Chromium side CL. https://chromium-review.googlesource.com/c/chromium/src/+/4456368

yoavweiss reviewed (approved):
LGTM
